cordova-lib/src/cordova/restore-util.js

```diff
@@ -112,7 +112,10 @@ function installPluginsFromConfigXML(args) {
         // assume it is the location to install from.
         var pluginSpec = pluginEntry.spec;
         pluginName = pluginEntry.name;
-        var installFrom = semver.validRange(pluginSpec, true) ? pluginName + '@' + pluginSpec : pluginSpec;
+        var installFrom = (pluginSpec && semver.validRange(pluginSpec, true)) ?
```

Comment:
This is a complex ternary - consider break into and if else as it will make it simpler to read.